ApolloTeleop.java

```diff
@@ -0,0 +1,226 @@
+
+
+        // Wait for the game to start (driver presses PLAY)
+        waitForStart();
+
+        // run until the end of the match (driver presses STOP)
+        while (opModeIsActive()) {
+
+            if (gamepad1.x) {
+                bibix = 2;
+                idle();
+            }
+
+            if (gamepad1.b) {
+                bibix = 1;
+                idle();
+            }
+
+            if (bibix == 1) {
+                right = gamepad1.right_stick_y;
+                left = gamepad1.left_stick_y;
+            }
+
+            if (bibix == 2) {
+                right = -gamepad1.right_stick_y;
+                left = -gamepad1.left_stick_y;
+            }
+
+            if (gamepad1.dpad_up) {
+                Doobi = 0.1;
+                Mcqueen = Mcqueen - 0.2;
+                while (gamepad1.dpad_up) {
+                    idle();
+                }
+                idle();
+            }
+
+
+            if (Doobi == 0.1) {
+                robot.leftDriveFront.setPower(left / Mcqueen);
+                robot.rightDriveFront.setPower(right / Mcqueen);
+                robot.leftDriveBack.setPower(left / Mcqueen);
+                robot.rightDriveBack.setPower(right / Mcqueen);
+                idle();
+            }
+
+
+            if (gamepad1.dpad_right) {
+                Doobi = 0.2;
+                idle();
+            }
+
+            if (Doobi == 0.2) {
+                robot.leftDriveFront.setPower(left / 1.7);
+                robot.rightDriveFront.setPower(right / 1.7);
+                robot.leftDriveBack.setPower(left / 1.7);
+                robot.rightDriveBack.setPower(right / 1.7);
+                Mcqueen = 1.7;
+                idle();
+            }
+
+            if (gamepad1.dpad_left) {
+                Doobi = 0.3;
+                idle();
+            }
+
+            if (Doobi == 0.3) {
+                robot.leftDriveFront.setPower(left / 2.5);
+                robot.rightDriveFront.setPower(right / 2.5);
+                robot.leftDriveBack.setPower(left / 2.5);
+                robot.rightDriveBack.setPower(right / 2.5);
+                Mcqueen = 2.5;
+                idle();
+            }
+
+
+            if (gamepad1.dpad_down) {
+                Doobi = 0.4;
+                Mcqueen += 0.2;
+                while (gamepad1.dpad_down) {
+                    idle();
+                }
+                idle();
+            }
+
+            if (Doobi == 0.4) {
+                robot.leftDriveFront.setPower(left / Mcqueen);
+                robot.rightDriveFront.setPower(right / Mcqueen);
+                robot.leftDriveBack.setPower(left / Mcqueen);
+                robot.rightDriveBack.setPower(right / Mcqueen);
+                idle();
+            }
+
+
+            if (gamepad2.right_trigger > 0) {
+                robot.lift.setPower(liftSpeed);
+            }
+
+            if (gamepad2.left_trigger > 0) {
+                robot.lift.setPower(-liftSpeed);
```

Comment:
You need to use the encoder to make sure that you are not trying to go more than you have cable for.
This can break the robot or motor.
Don't try it before you fix this part!